### DISCUSSION TITLE
Libre2 direct - BT Pairing goes to "Not Bonded" & 2 hours missed readings

### DISCUSSION THREAD

**Discussion by nigelkeirross** [NightscoutFoundation/xDrip#2848] [Q&A]
Last night @ 0335 I woke to discover that I had over 2 hours of missed readings
on xDrip+. I am using libre2 (UK) direct, and have had ~ 3 days of trouble-free operation.
When I looked at the Stateus page (BT), I noticed that it was disconnected, and would try to reconnect. I saw it connect about 3 times, but then immediately ( < 1 sec) disconnect again.
I then did an NFC scan which was OK, but still disconnected.
Then I noticed that the "Bluetooth Pairing" value was "Not Bonded (3)",
and that the "next retry" value would sometimes be negative.
After a few minutes I tried tapping on the "Bluetooth Pairing", and it reverted to the normal "Disabled, tap to enable", and then it connected & I got readings.
For about 30 minutes, there were unusual events shown in the log, but it then reverted to normal behaviour.
I am wondering what could have put xDrip into this mode - I was asleep at the time, so there were no manual changes that I did.
![033600-notbonded-3-negative-next-retry](https://github.com/NightscoutFoundation/xDrip/assets/21008320/a64ef05e-4faf-4e4a-8295-c1fa17487e54)
![034116-eventlog](https://github.com/NightscoutFoundation/xDrip/assets/21008320/801d2a6c-b698-4829-a31f-3f6ddd9cbbe2)

**Comment by Navid200** [comment]
@psonnera

**Reply by psonnera** [reply]
@nigelkeirross at 3:40 see that glucoseLevel = -13? That's a sensor error. This measurement time stamp was 2401 but was removed due to the error. To me it looks like this missing value is carried away and even if next values are read correctly (as OOP shows) it does disturb something.
On the other hand, at 3:38 there is a blutooth error. I suspect this is why you lost connection but don't know if it's related to the error above.
Did you disable trust auto-connect and both scanning options in xDrip+ bluetooth settings?

**Reply by nigelkeirross** [reply]
@psonnera My BT settings are: Use Scanning - off, Trust Auto-connect - ON, Use Background scans ON. I also see I have "Allow Blucon unbonding ON. 
The problem started at 0129, as can be seen from the log. Up till then, all had been good & then stopped getting readings...
![log-whenitwentwrong](https://github.com/NightscoutFoundation/xDrip/assets/21008320/0a7143f3-0656-40b9-9ff9-83f58cbbb3d7)

**Comment by parapenT1sta** [comment]
It seems you aren't connected directly to Libre 2 sensor. Do you have Bluetooth device: ABBOTT... in the "Classic Status Page" tab?

**Reply by nigelkeirross** [reply]
Yes I have ABBOTT... as the Bluetooth device. My point is that the sensor had been working happily for a couple of days, but then something happened in the middle of the night while I was asleep to cause me to miss 2 hours of readings, and leave the sensor disconnected, and xDrip+ in the state as described. I am hoping to learn what could have caused this, as it seems that it got into a state where it would never connect again unless I had tapped on the Bluetooth Pairing line.

**Reply by PhilTU** [reply]
I haven't had this happen. Have seen people with the problem but usually they had been trying everything in sight and hit it manually.
